integration-tests/impersonate/impersonate_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 Google LLC.
```

Comment:
what is this directory "integration-tests"? does it need to be a top-level directory?

Reply:
It already exists today. It holds some tests for idtoken and storage today. Maybe in a future PR we should move it to internal?